Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3528 @@
+                file_indicator.dbot_score = file_indicator.dbot_score or dbot_score
+
+                is_all_filled = file_indicator.md5 \
+                    and file_indicator.sha1 \
+                    and file_indicator.sha256 \
+                    and file_indicator.path \
+                    and file_indicator.name \
+                    and file_indicator.hostname \
+                    and file_indicator.dbot_score
+
+                if is_all_filled:
+                    break
+
+            file_context = file_indicator.to_context().get(Common.File.CONTEXT_PATH)
+            readable_output = tableToMarkdown(
+                f'Cisco AMP - Hash Reputation for: {file_hash}',
+                file_context
+            )
+
+            command_results.append(CommandResults(
+                readable_output=readable_output,
+                outputs_prefix='',
+                raw_response=raw_response,
+                outputs_key_field='SHA256',
+                indicator=file_indicator
+            ))
+
+        except Exception as exc:  # pylint: disable=broad-except
+            command_results.append(CommandResults(
+                readable_output=f'Could not process file: "{file_hash}"\n{str(exc)}')
+            )
+
+    return command_results
+
+
+''' HELPER FUNCTIONS '''  # pylint: disable=pointless-string-statement
+
+
+def get_pagination_parameters(
+    page: int = None,
+    page_size: int = None,
+    limit: int = None
+) -> Pagination:
+    """
+    Get the limit and offset required for the http request,
+    number of requests required and if the pagination is automatic, manual or none of them.
+
+    Args:
+        page (int, optional): Page number to view. Defaults to None.
+        page_size (int, optional): Number of elements in each page. Defaults to None.
+        limit (int, optional): Total number of elements to return. Defaults to None.
+
+    Raises:
+        ValueError: If both manual and automatic arguments have been filled.
+
+    Returns:
+        Pagination:
+            page (int): Page number to view.
+            page_size (int): Number of elements in each page.
+            limit (int): Total number of elements to return.
+            offset (int): Number of 1st element to return.
+            number_of_requests (int): Number of http requests to make.
+            offset_multiplier (int): Multiply by 1 when number of requests is 1, otherwise 0.
+            is_automatic (bool): Whether the pagination type is automatic.
+            is_manual (bool): Whether the pagination type is manual.
+    """
+    is_automatic: bool = limit is not None
+    is_manual: bool = page is not None or page_size is not None
+
+    if is_manual and is_automatic:
+        raise ValueError('page or page_size can not be entered with limit.')
+
+    # Automatic Pagination
+    if is_automatic:
+        if limit > MAX_PAGE_SIZE:  # type: ignore
```

Comment:
why `type:ignore`?

Reply:
I've got an operator error by mypy, because limit is `Optional[int]`.
This isn't possible, because `is_automatic` holds `limit is not None`.